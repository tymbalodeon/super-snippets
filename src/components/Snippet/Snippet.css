.Snippet_wrapper a {
  text-decoration: none;
}

.Snippet__details h2 {
  color: #5ccfe6;
}

.Snippet__text p {
  color: #bae67e;
}
