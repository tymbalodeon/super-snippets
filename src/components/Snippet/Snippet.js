import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Snippet.css';

export default class Snippet extends Component {
  render() {
    const { snippet } = this.props;
    return (
      <div className="Snippet_wrapper">
        <Link to={`/snippets/${snippet.id}`}>
          <div className="Snippet__details">
            <div className="Snippet__text">
              <h2 className="Snippet__heading">{snippet.snippet_name}</h2>
              <p className="Snippet__description">{snippet.info}</p>
            </div>
          </div>
        </Link>
      </div>
    );
  }
}
