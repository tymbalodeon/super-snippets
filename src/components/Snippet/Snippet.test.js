import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Snippet from './Snippet';
import testSeed from '../../test-seed';

describe('<Snippet />', () => {
  const snippet = testSeed.makeSnippet();

  it('renders without crashinS', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <Snippet snippet={snippet} />
      </BrowserRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Snippet snippet={snippet} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
