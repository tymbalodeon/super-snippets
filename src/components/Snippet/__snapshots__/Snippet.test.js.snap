// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snippet /> renders the UI as expected 1`] = `
<div
  className="Snippet_wrapper"
>
  <a
    href="/snippets/1"
    onClick={[Function]}
  >
    <div
      className="Snippet__details"
    >
      <div
        className="Snippet__text_small"
      >
        <h2
          className="Snippet__heading"
        />
        <p
          className="Snippet__description"
        >
          testing stuff
        </p>
      </div>
    </div>
  </a>
</div>
`;
